File: helps.py
class Helper:
    def __init__(self, rg: str, args: dict):
        self.args = args
        helpSet = {
            "puts": self.Puts,
            "list": self.Ls,
            "remove": self.Remove,
            "unpack": self.Unpack,
            "zipto": self.Zipto,
            "*": self.All,
        }
        helpSet[rg]()

    def Puts(self):
        print("\n*  novem PUTS [ PATH ]")
        print("\t        : logs the given PATH(s) as a dotfile")
        print("\t        : (this will also overwrite an existing dot if novem already logged it)")
        print("\n")

    def Ls(self):
        print("\n*  novem LIST [ {string}, here ]")
        print("\t none     : lists all dotfiles, details depend on verbosity")
        print("\t here     : tries to see if any logged dots are in current dir")
        print("\t{string}  : searches for {str} in dotfiles")

    def Remove(self):
        print("\n*  novem REMOVE [ PATH ]")
        print("\t        : removes PATH from logs")
        print("\t        : (after confirmation)")

    def Unpack(self):
        print("\n*  novem UNPACK")
        print("\t(no args) : unpacks all aliases to DEFAULT_SHELL_SOURCE_PATH")
        print(f"\t>>> SOURCE_PATH : {self.args['conf']}")

    def Zipto(self):
        print("\n*  novem UPDATE [ alias, type ]")
        print("\t'alias' : prompts for & edits alias content")
        print("\t'type'  : prompts for & alters type name in DEFAULT_CONFIG_DIR_PATH")
        print(f"\t>>> SOURCE_PATH : {self.args['zip']}")

    def All(self):
        self.Puts()
        self.Ls()
        self.Remove()
        self.Unpack()
        self.Zipto()

File: helps.go
package main

import (
	"fmt"
	"strings"
)

func BlankLine() {
	s := "\n\n"
	fmt.Printf("%s", s)
}

func EvalHelp(h string) {
	s := novemIcon + "    HELP"
	fmt.Println(s)
	fmt.Println(strings.Repeat("  -", len(s)/2))
	switch h {
	case "puts":
		PutsHelp()

	case "list":
		LsHelp()

	case "remove":
		RemoveHelp()

	case "unpack":
		UnpackHelp()

	case "zipto":
		ZiptoHelp()
	}
}

func PutsHelp() {
	fmt.Println(novemIcon, "  novem PUTS [ PATH ]")
	fmt.Println("\t        : logs the given PATH(s) as a dotfile")
	fmt.Println("\t        : (this will also overwrite an existing dot if novem already logged it)")
	BlankLine()
}

func LsHelp() {
	fmt.Println(novemIcon, "  novem LIST [ path ]")
	fmt.Println("\t         : lists all dotfiles, details depend on verbosity")
	fmt.Println("\t         : uses current dir if no specific path given")
	fmt.Println("\t{string} : searches for {str} in dotfiles")
	BlankLine()
}

func RemoveHelp() {
	fmt.Println(novemIcon, "  novem REMOVE [ PATH ]")
	fmt.Println("\t        : removes PATH from dotfiles")
	fmt.Println("\t          (after confirmation)")
	BlankLine()
}

func UnpackHelp() {
	fmt.Println(novemIcon, "  novem UNPACK [ .zip file ]")
	fmt.Println("\t          : attempts to unpack a .zip into a Novem json")
	fmt.Println("\t     (if valid, unpacks all dotfiles to their respective PATHs)")
	BlankLine()
}

func ZiptoHelp() {
	fmt.Println(novemIcon, "  novem ZIPTO  (optional zip output path)")
	fmt.Printf("\t>>> DEFAULT NOVEM ZIP PATH:  %s", PATH)
	BlankLine()
}

func PrintAllHelp() {
	PutsHelp()
	LsHelp()
	RemoveHelp()
	UnpackHelp()
	ZiptoHelp()
}
